src/providers/hollisterco/scraper.ts

```diff
@@ -0,0 +1,174 @@
+        el => ({
+          ...Object.assign({}, el.dataset),
+        }),
+      )
+    } else {
+      // @ts-ignore
+      onlyOneAvailable = options.find(o => o.inventoryStatus === 'Available').swatch
+      swatch = [
+        {
+          // @ts-ignore
+          productid: Object.keys(productCatalog)[0],
+          swatch: onlyOneAvailable,
+          producturl: null,
+        },
+      ]
+    }
+
+    swatch = swatch.map(sw => {
+      //@ts-ignore
+      let p = productCatalog[sw.productid]
+      let kv = p.productAttrsComplex.FiberContent
+      let keyValues = {}
+
+      if(kv !== undefined) {
+        // check for .value (str) or .values []
+        if (kv.value) {
+          let [key, value] = kv.value.split(':')
+          keyValues[key] = value
+        } else {
+          //@ts-ignore
+          for (let pairs of p.productAttrsComplex.FiberContent.values) {
+            let [key, value] = pairs.value.split(':')
+            keyValues[key] = value
+          }
+        }
+      }
+
+      let bullets = p.productAttrsComplex.CareInstructions?.values.map(o => o.value)
+      bullets = Array.isArray(bullets) ? bullets : []
+
+      return {
+        //@ts-ignore
+        ...sw,
+        sizeChartUrls: [
+          `https://www.hollisterco.com/api/ecomm/h-wd/product/sizeguide/${p.sizeChartName}`,
+        ],
+        description: p.longDesc,
+        keyValuePairs: keyValues,
+        bullets: bullets,
+        //@ts-ignore
+        images: Object.values(productCatalog[sw.productid].imageSets)
+          .flat()
+          //@ts-ignore
+          .filter(o => o.id)
+          //@ts-ignore
+          .map(o => `https://img.hollisterco.com/is/image/anf/${o.id}?policy=product-large`),
+      }
+    })
+
+    options = options
+      .filter((op: any) => !onlyOneAvailable || op.swatch === onlyOneAvailable)
+      .map((op: any) => ({
+        ...op,
+        ...swatch.find(s => s.swatch === op.swatch),
+      }))
+
+    options = options.map((op: any) => {
+      //@ts-ignore
+      let item:any = Object.values(productPrices[op.productid].items)[0]
+      return {
+        ...op,
+        ...item
+      }
+  })
+
+    return {
+      options,
+      title,
+      subtitle,
+      breadcrumbs,
+      currency,
+    }
+  })
+
+  data.options = data.options.map((op: any) => ({
+    ...op,
+    producturl:
+      op.producturl === null ? itemGroupUrl : new URL(itemGroupUrl).hostname + op.producturl,
+  }))
+
+  console.dir(data, { depth: null })
+
+  const products = data.options.map((v: any) => {
+    let p = new Product(`${v.idx}-${v.productid}`, data.title, v.producturl)
+
+    p.subTitle = data.subtitle
+    p.metadata = {}
+    p.images = v.images
+    p.videos = []
+    p.description = v.description
+    p.currency = data.currency
+    p.sku = v.sku
+    p.brand = v.brand
+    p.size = v.sizePrimary + (v.sizeSecondary ? ` - ${v.sizeSecondary}` : '')
+    p.realPrice = v.offerPrice
```

Comment:
Just in case. Generalmente es mas seguro suponer que cuando no hay oferta puede venir empty el dato, por lo que se podría contemplar de la siguiente manera:

```typescript
p.realPrice = v.offerPrice || v.listPrice